src/main/org/deidentifier/arx/ARXConfiguration.java

```diff
@@ -707,7 +707,9 @@ public Monotonicity getMonotonicityOfPrivacy() {
      * @return
      */
     public Monotonicity getMonotonicityOfUtility() {
-        if (metric.isMonotonic() || (this.getMaxOutliers() == 0d) || this.isPracticalMonotonicity()) {
+        if ((metric.isMonotonicWithGeneralization() && this.getMaxOutliers() == 0) || 
```

Comment:
The action code is here (plus equivalent in lattice and lightning).  My understanding is that generalization and suppression are mutually exclusive when associated with number of outliers, so all evaluations are setup as ((&&) || (&&)).

In my first change, I actually bungled this logic and testing by comparison of 3.4 to 3.5 failed.  I was a bit surprised that unit tests did not reflect the issue (all passed).  Would tests for a specific generalization scheme be in order?  Seems like the existing utility tests are based on information loss?